### PR TITLE
test: add availability booking e2e coverage (AT-004)

### LINKED ISSUE #487: [TASK]: Implement Acceptance Test for Availability & Booking (AT-004)
## Description

**Detailed Information:**
Implement the Playwright acceptance test for slot management and booking workflows.

## Motivation

**Reasoning and Purpose:**
To validate the end-to-end collaborative booking workflow, ensuring mentors can publish availability without conflicts and mentees can successfully discover and book these slots.

## Scope

**Boundaries and Extent:**
- Write Playwright E2E tests covering availability management for AT-004.
- Validate slot creation, overlap prevention, booking requests (with optional cover letters), and acceptance workflows.

## Acceptance Criteria

- [x] Acceptance test is implemented using Playwright
- [x] Test passes when run locally
- [x] Reviewed by a team member

## Notes
Please refer to the [Acceptance Tests Checklist](https://github.com/bounswe/bounswe2026group5/wiki/Acceptance-Tests-Checklist) — Availability & Booking section for the exact step-by-step instructions and expectations.

### PR DESCRIPTION
**Pull request by EbubekirErden**
## Related Issue

Closes #487 

## Description

This PR adds Playwright E2E coverage for the Availability & Booking acceptance scenario (`AT-004`).

Key changes:
- Adds a POM-based Availability & Booking E2E test.
- Adds `DashboardPage`, `DiscoverPage` and `ProfilePage` page objects.
- Adds `TestDataApi` for synthetic user setup, browser login, and discovery API verification.
- Covers the full mentor–mentee booking flow, including availability publishing, slot discovery and booking, user feedback, and safeguards against expired, duplicate, or unauthorized booking requests.

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Refactoring (no functional changes)
- [ ] Documentation
- [x] CI/CD or configuration

## Checklist

- [x] I have tested my changes locally
- [x] My code builds without errors
- [x] I have written or updated tests where applicable
- [x] I have performed a self-review of my code

## Notes

This branch is stacked on top of the Playwright setup branch. The test is web-only and follows the POM direction used by the authentication E2E work.

### CODE REVIEW

File: web/e2e/availability-booking.spec.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      skills: ['Machine Learning'],
+    };
+    const coverLetter = `Hi Dr Ayse, I would like help planning a machine learning project. Run ${runId}.`;
+
+    const mentorAuth = await api.seedUser(mentor, 'MENTOR');
+    const menteeAuth = await api.seedUser(mentee, 'MENTEE');
+    const otherMenteeAuth = await api.seedUser(otherMentee, 'MENTEE');
+
+    const slotDate = toDateString(new Date(Date.now() + 24 * 60 * 60 * 1000));
+    const firstSlot = await api.createAvailabilitySlot(mentorAuth, {
```
Comment (Cydonia01) [inline web/e2e/availability-booking.spec.ts]: You should first check that there is no availability slot (Step 1). Then you can add and check if it exists (Step 2).

**Review by Cydonia01 — CHANGES_REQUESTED** [review]
Also, steps 3,4 and 5 also missing. Watch out for the edge cases. They are important.

**Comment by EbubekirErden** [comment]
@Cydonia01 I updated the test to cover the missing early availability steps: empty calendar, slot creation, overlap rejection, adjacent slot creation, and past-date rejection. Some steps are adapted to the current web UI: slot creation is done by clicking calendar cells instead of an “Add Slot” form, and overlap/past-date checks are verified through API because the UI does not support arbitrary time entry or past-date selection. Could you review it again when you have time?

**Review by Cydonia01 — APPROVED** [review]
Now, it looks more detailed and fulfills the test specification in [Acceptance Tests Strategy](https://github.com/bounswe/bounswe2026group5/wiki/Acceptance-Tests-Checklist) wiki page. Thank you for your efforts.
